### PR TITLE
Issue3316

### PR DESCRIPTION
**Pull request by corinnekrych**
Fixes https://github.com/openshiftio/openshift.io/issues/3316

> NOTE: merge this https://github.com/xcoulon/unleash-client-go/pull/1 then we can modify Gopkg.toml|lock to keep pointing on Xavier fork 

Once this PR is merged we can work on removing hard-coded list of features in the ui

### CODE REVIEW

**Comment by corinnekrych** [comment]
```
curl http://localhost:8080/api/features\?strategy\=enableByEmails -H "Authorization: Bearer $TOKEN"
{"data":[{"attributes":{"description":"new home dashbaord","enabled":true,"user-enabled":false},"id":"Analyze.newHomeDashboard","type":"features"}]}

curl http://localhost:8080/api/features\?strategy\=enableByEmails                                  
{"data":[{"attributes":{"description":"new home dashbaord","enabled":true,"user-enabled":false},"id":"Analyze.newHomeDashboard","type":"features"}]}

curl http://localhost:8080/api/features\?strategy\=enableByLevel -H "Authorization: Bearer $TOKEN" 
{"data":[{"attributes":{"description":"analyse","enabled":true,"enablement-level":"released","user-enabled":true},"id":"Analyze","type":"features"},{"attributes":{"description":"ddd","enabled":true,"enablement-level":"internal","user-enabled":true},"id":"Analyze.newSpaceDashboard","type":"features"},{"attributes":{"description":"eeee","enabled":true,"enablement-level":"released","user-enabled":true},"id":"Planner","type":"features"}]}
```
Some curl to test endpoint ^^

File: controller/features_test.go [hunk 1]
```diff
@@ -497,14 +503,40 @@ func TestListFeatures(t *testing.T) {
...
 			// then
 			expectedData := []*app.UserFeature{}
 			assert.Equal(t, expectedData, featuresList.Data)
 		})
 
 	})
 
+	t.Run("list by strategy", func(t *testing.T) {
+
+		t.Run("1 match", func(t *testing.T) {
```
Comment (xcoulon) [inline controller/features_test.go]: could you add a test in which there's no match ? I assume the response would be `200 OK` with an empty list ?

**Comment by corinnekrych** [comment]
@xcoulon I've integrated your comment, 
mind another review?
